Remove apache compress dependency

DavyLandman opened the pull request:
We had this full dependency that we only used for 2 classes, that are in reality not used, and could be replaced by existing dependencies.

jurgenvinju reviewed (approved):
I assume we have tests for this code.

jurgenvinju commented:
pls merge at will

DavyLandman commented:
I'm not sure this is ever tested, let me check.

jurgenvinju commented:
It's of particular importance that files written with the previous version can be read with the new one.

DavyLandman commented:
The underlying library used are the exact same, I just removed the apache-compress abstraction layer. So this is not an issue of not being able to read the files anymore. I just realized that the different compression modes weren't even tested. I've added a new PR (#247) that introduces them with regards to main first.

DavyLandman commented:
Ok, I've updated the branch to be rebased on-top of main. so it should now run the new compression tests.

jurgenvinju commented:
I saw that, but I wanted to test it too.

On Wed, Feb 28, 2024 at 9:36 PM Davy Landman ***@***.***>
wrote:

> The underlying library used are the exact same, I just removed the
> apache-compress abstraction layer. So this is not an issue of not being
> able to read the files anymore. I just realized that the different
> compression modes weren't even tested. I've added a new PR (#247
> <https://github.com/usethesource/vallang/pull/247>) that introduces them
> with regards to main first.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/usethesource/vallang/pull/245#issuecomment-1969875363>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAPF5FZV5VTUYDXMNGBUOC3YV6INJAVCNFSM6AAAAABDS3BPECVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNRZHA3TKMZWGM>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>


-- 
Jurgen Vinju
  http://jurgen.vinju.org
  CWI SWAT
  TU Eindhoven
  Swat.engineering BV

jurgenvinju commented:
no problem! I never expected a failure, but I just wanted to test once
before we release. I did that this afternoon and we're ok. It works like a
charm on old files.

On Thu, Feb 29, 2024 at 7:26 PM Davy Landman ***@***.***>
wrote:

> Ah, you wrote: merge when you feel for it. That's why I merged.
>
> On Thu, 29 Feb 2024, 19:24 Jurgen J. Vinju, ***@***.***>
> wrote:
>
> > I saw that, but I wanted to test it too.
> >
> > On Wed, Feb 28, 2024 at 9:36 PM Davy Landman ***@***.***>
> > wrote:
> >
> > > The underlying library used are the exact same, I just removed the
> > > apache-compress abstraction layer. So this is not an issue of not
> being
> > > able to read the files anymore. I just realized that the different
> > > compression modes weren't even tested. I've added a new PR (#247
> > > <https://github.com/usethesource/vallang/pull/247>) that introduces
> > them
> > > with regards to main first.
> > >
> > > —
> > > Reply to this email directly, view it on GitHub
> > > <
> > https://github.com/usethesource/vallang/pull/245#issuecomment-1969875363>,
>
> >
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AAPF5FZV5VTUYDXMNGBUOC3YV6INJAVCNFSM6AAAAABDS3BPECVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNRZHA3TKMZWGM>
>
> >
> > > .
> > > You are receiving this because your review was requested.Message ID:
> > > ***@***.***>
> > >
> >
> >
> > --
> > Jurgen Vinju
> > http://jurgen.vinju.org
> > CWI SWAT
> > TU Eindhoven
> > Swat.engineering BV
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/usethesource/vallang/pull/245#issuecomment-1971709274>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AABL3E4YZQDDO35YCPWVBZLYV5Y65AVCNFSM6AAAAABDS3BPECVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNZRG4YDSMRXGQ>
>
> > .
> > You are receiving this because you modified the open/close state.Message
> > ID: ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/usethesource/vallang/pull/245#issuecomment-1971711007>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAPF5FYL5C53543FGBR3EYLYV5ZDPAVCNFSM6AAAAABDS3BPECVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNZRG4YTCMBQG4>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>


-- 
Jurgen Vinju
  http://jurgen.vinju.org
  CWI SWAT
  TU Eindhoven
  Swat.engineering BV

DavyLandman commented:
Just to be clear, these two compressors are never used in rascal, as we always use normal mode. 

So this was unused code (and to be honest, it crashed when you would trigger it, the other pr fixed that).
